uc-crux-llvm/src/UCCrux/LLVM/Classify.hs

```diff
@@ -0,0 +1,557 @@
+                           unfixed appCtx tag
+                       _ ->
+                         case What4.asConcrete offset of
+                           Just bv ->
+                             do
+                               let tag = ArgPointerConstOffset
+                               liftIO $
+                                 (appCtx ^. log) Hi $
+                                   Text.unwords
+                                     [ "Diagnosis: ",
+                                       diagnose tag,
+                                       "#" <> Text.pack (show (Ctx.indexVal idx)),
+                                       "at",
+                                       Text.pack (show (ppCursor (argName idx) cursor)),
+                                       "(" <> Text.pack (show (LLVMPtr.ppPtr ptr)) <> ")"
+                                     ]
+                               liftIO $ (appCtx ^. log) Hi $ prescribe tag
+                               case ftRepr of
+                                 FTPtrRepr partType ->
+                                   return $
+                                     ExMissingPreconditions
+                                       ( tag,
+                                         oneArgShapeConstraint
+                                           idx
+                                           cursor
+                                           (Allocated (fromIntegral (elemsFromOffset' bv partType)))
+                                       )
+                                 _ -> panic "classify" ["Expected pointer type"]
+                           Nothing ->
+                             do
+                               let tag = AddSymbolicOffsetToArgPointer
+                               liftIO $
+                                 (appCtx ^. log) Hi $
+                                   Text.unwords
+                                     [ "Diagnosis:",
+                                       ppUnfixable tag,
+                                       "#" <> Text.pack (show (Ctx.indexVal idx)),
+                                       "at",
+                                       Text.pack (show (ppCursor (argName idx) cursor)),
+                                       "(" <> Text.pack (show (LLVMPtr.ppPtr ptr)) <> ")"
+                                     ]
+                               unfixable appCtx tag
+                   _ -> unclass appCtx badBehavior
+             LLVMErrors.BBUndefinedBehavior
+               (UB.MemsetInvalidRegion (Crucible.RV ptr) _fillByte (Crucible.RV len)) ->
+                 case (getPtrOffsetAnn ptr, What4.asConcrete len) of
+                   ( Just
+                       ( Some
+                           ( TypedSelector
+                               ftRepr
+                               (SomeInSelector (SelectArgument idx cursor))
+                             )
+                         ),
+                     Just concreteLen
```

Comment:
I've tried to really roughly keep lines of manageable length (generally <~100 columns). There are a few others patterns (`NoSatisfyingWrite`) that are broken across lines, usually because there's something being `case`d on in addition to the selector.